DynamicTablesPkg/Library/Acpi/Common/AcpiMcfgLib/McfgGenerator.c

```diff
@@ -162,12 +173,21 @@ BuildMcfgTable (
              );
   if (EFI_ERROR (Status)) {
     DEBUG ((
-      DEBUG_ERROR,
-      "ERROR: MCFG: Failed to get PCI Configuration Space Information." \
-      " Status = %r\n",
-      Status
+      DEBUG_INFO,
+      "INFO: MCFG: PCI Configuration Space Information not provided by platform." \
+      "          : Retrieving PCI configuration space details...\n"
       ));
-    goto error_handler;
+    PciSegmentInfo = GetPciSegmentInfo (&PciConfigurationSpaceCount);

```

Comment:
The Configuration Manager is responsible for collating the hardware information and populating the Platform Info Repository. Therefore, I am wondering if this can be done as part of the Configuration Manager, i.e. the Configuration Manager can call GetPciSegmentInfo and populate the PlatformInfo Repository with EArchCommonObjPciConfigSpaceInfo objects.

Otherwise I think we are breaking the design. Any thoughts?

Reply:
Objective of this patch is to minimize the platform configuration.
So that it will be easy to port the changes across different platforms and boards.

GetPciSegmentInfo() is platform independent function and gives the same PCI segment information.
Whereas platform configuration driver should (in my opinion) contains the data which will be different across platform.

Also this patch wont break(or alter) the existing behavior.